app/views/layouts/application.html.erb

```diff
@@ -1,21 +1,119 @@
 <!DOCTYPE html>
 <html>
 <head>
+
   <title>Blog</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <%= stylesheet_link_tag    'application', media: 'all' %>
+  <%= javascript_include_tag 'application' %>
   <%= csrf_meta_tags %>
 </head>
 <body>
 
-<%= link_to "Home", root_path %>
-<%= link_to "About", about_path %>
-<%= link_to "Contact Us", contact_path %>
-<br>
-<hr>
+    <!-- Nav Bar -->
+    <nav class="navbar navbar-default my-navbar">
+      <div class="container">
+        <!-- Brand and toggle get grouped for better mobile display -->
+        <div class="navbar-header">
+          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
+            <span class="sr-only">Toggle navigation</span>
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+          </button>
+          <%= image_tag "toms_logo_trans.png", class: "brand-image" %>
+        </div>
+
+        <!-- Collect the nav links, forms, and other content for toggling -->
+        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+          <ul class="nav navbar-nav">
+            <li><%= link_to "Home", root_path %></li>
+            <%# modal pop-up for contact us %>
+            <li><%= link_to "Contact Us", contacts_path, remote: true, method: :post, data: {toggle: "modal", target: "#contactModal"} %></li>
+          </ul>
+
+          <!--CHANGE THESE LINKS IF USER IS SIGNED IN / SIGNED UP OR NOT -->
+          <!--replace rails link -->
+          <ul class="nav navbar-nav navbar-right">
+            <!--display this nav bar if the user is signed in -->
+            <% if user_signed_in? %>
+              <li class="welcome-text"><em>Welcome Back:  <%= current_user.show_name %></em></li>
+              <li><%= link_to "Create A Post", new_post_path, class: "btn btn-success my-btn" %></li>
+              <li><%= link_to "View Favorites", favorites_path, class: "btn btn-primary my-btn" %></li>
+              <li><%= link_to "Sign Out", destroy_user_session_path, method: :delete, class: "btn btn-danger my-btn" %></li>
+            <% else %>
+              <li><%= link_to "Sign In", new_user_session_path, class: "btn btn-success my-btn" %></li>
+              <li><%= link_to "Sign Up", new_user_registration_path, class: "btn btn-primary my-btn" %></li>
+            <% end %>
+          </ul>
+        </div><!-- /.navbar-collapse -->
+      </div><!-- /.container-fluid -->
+    </nav>
+
+
+    <!--Hero -->
```

Comment:
could put the hero-wrapper in a partial or in the index file instead of having separate files for devise.html.erb and applicaiton.html.erb